lib/common/observer/observer.go

```diff
@@ -1,9 +1,74 @@
 package observer
 
-import "github.com/GianlucaGuarini/go-observable"
+import (
+	"github.com/GianlucaGuarini/go-observable"
+)
 
 var BlockAccountObserver = observable.New()
 var BlockTransactionObserver = observable.New()
 var BlockObserver = observable.New()
 var BlockOperationObserver = observable.New()
 var SyncBlockWaitObserver = observable.New()
+
+var ResourceObserver = observable.New()
+
+const (
+	ResourceTransaction     = "tx"
+	ResourceTransactionPool = "txpool"
+	ResourceOperation       = "op"
+	ResourceAccount         = "ac"
+	ConditionAll            = "*"
+	ConditionSource         = "source"
+	ConditionTarget         = "target"
+	ConditionType           = "type"
+	ConditionOpHash         = "ophash"
+	ConditionTxHash         = "txhash"
+	ConditionAddress        = "address"
+)
+
+type Event struct {
+	Resource  string `json:"resource"`
+	Condition string `json:"condition"`
+	Id        string `json:"id"`
+}
+
+func NewEvent(resource, condition, id string) Event {
+	return Event{
+		Resource:  resource,
+		Condition: condition,
+		Id:        id,
+	}
+}
+func (e Event) String() string {
+	toStr := e.Resource + "-"
+	if e.Condition == ConditionAll {
+		toStr += e.Condition
+	} else {
+		toStr += e.Condition + "="
+		toStr += e.Id
+	}
+	return toStr
+}
+
+type Subscribe struct {
```

Comment:
Yes it is.
And it is used to trigger the events.

Reply:
If it is, subscribe should be used with api or http layer. IMHO, the Subscribe can be in httputils :-)

Reply:
IMHO, it is closer to `observer` than `httputils`
because the Subscribe, Event can be used not only api but also other modules
If needed, there can be another structure for httputils which converted to the Subscribe and Event

Reply:
This object just makes event's string, It doesn't do subscribe..